docs/notebooks/convolutions.md

```diff
@@ -0,0 +1,506 @@
+
++++ {"id": "ewZEn2X12-Ng"}
+
+## Basic One-dimensional Convolution
+
+Basic one-dimensional convolution is implemented by {func}`jax.numpy.convolve`, which provides a JAX interface for {func}`numpy.convolve`. Here is a simple example of 1D smoothing implemented via a convolution:
+
+```{code-cell}
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 265
+id: 0qYLpeZO3Z9-
+outputId: 4f6717ac-a062-4a85-8330-d57bf80de384
+---
+import matplotlib.pyplot as plt
+
+from jax import random
+import jax.numpy as jnp
+import numpy as np
+
+key = random.PRNGKey(1701)
+
+x = jnp.linspace(0, 10, 500)
+y = jnp.sin(x) + 0.2 * random.normal(key, shape=(500,))
+
+window = jnp.ones(10) / 10
+y_smooth = jnp.convolve(y, window, mode='same')
+
+plt.plot(x, y, 'lightgray')
+plt.plot(x, y_smooth, 'black');
+```
+
++++ {"id": "dYX1tCVB4XOW"}
+
+The `mode` parameter controls how boundary conditions are treated; here we use `mode='same'` to ensure that the output is the same size as the input.
+
+For more information, see the {func}`jax.numpy.convolve` documentation, or the documentation associated with the original {func}`numpy.convolve` function.
+
++++ {"id": "5ndvLDIH4rv6"}
+
+## Basic N-dimensional Convolution
+
+For *N*-dimensional convolution, {func}`jax.scipy.signal.convolve` provides a similar interface to that of {func}`jax.numpy.convolve`, generalized to *N* dimensions.
+
+For example, here is a simple approach to de-noising an image based on convolution with a Gaussian filter:
+
+```{code-cell}
+---
+colab:
+  base_uri: https://localhost:8080/
+  height: 212
+id: Jk5qdnbv6QgT
+outputId: 292205eb-aa09-446f-eec2-af8c23cfc718
+---
+from scipy import misc
+import jax.scipy as jsp
+
+fig, ax = plt.subplots(1, 3, figsize=(12, 5))
+
+# Load a sample image; compute mean() to convert from RGB to grayscale.
+image = jnp.array(misc.face().mean(-1))
+ax[0].imshow(image, cmap='binary_r')
+ax[0].set_title('original')
+
+# Create a noisy version by adding random Gausian noise
+key = random.PRNGKey(1701)
+noisy_image = image + 50 * random.normal(key, image.shape)
+ax[1].imshow(noisy_image, cmap='binary_r')
+ax[1].set_title('noisy')
+
+# Smooth the noisy image with a 2D Gaussian smoothing kernel.
+x = jnp.linspace(-3, 3, 7)
+window = jsp.stats.norm.pdf(x) * jsp.stats.norm.pdf(x[:, None])
+smooth_image = jsp.signal.convolve(noisy_image, window, mode='same')
+ax[2].imshow(smooth_image, cmap='binary_r')
+ax[2].set_title('smoothed');
+```
+
++++ {"id": "Op-NhXy39z2U"}
+
+Like in the one-dimensional case, we use `mode='same'` to specify how we would like edges to be handled. For more information on available options in *N*-dimensional convolutions, see the {func}`jax.scipy.signal.convole` documentation.
+
++++ {"id": "bxuUjFVG-v1h"}
+
+## General Convolutions
```

Comment:
These are copy/pasted from the Sharp Bits notebook right? As someone who doesn't know much about convolutions, I find these examples confusing, but maybe I'm not the target audience and/or we should land this as a copy/paste and make improvements from there? If you do wanna fix this section up, lemme know and I can leave more detailed review comments.

Reply:
Yes, this is just directly copy-pasted. I might make a pass at these in a later PR, but I'd prefer to leave them unchanged for now.